testing/matrix/linter/rules/rules.go

```diff
@@ -166,24 +171,68 @@ func containerEnvVariablesDuplicates(object storage.StoreObject, containers []v1
 	return errors.EmptyRuleError
 }
 
-func containerImageTagLatest(object storage.StoreObject, containers []v1.Container) errors.LintRuleError {
+func shouldSkipModuleContainer(module string, container string) bool {
+	// okmeter module uses images from external repo - registry.okmeter.io/agent/okagent:stub
+	if module == "okmeter" && container == "okagent" {
+		return true
+	}
+	// control-plane-manager uses `$images` as dict to render static pod manifests,
+	// so we cannot use helm lib `helm_lib_module_image` helper because `$images`
+	// is also rendered in `dhctl` tool on cluster bootstrap.
+	if module == "d8-control-plane-manager" && strings.HasPrefix(container, "image-holder") {
+		return true
+	}
+	return false
+}
+
+func containerImageTagCheck(object storage.StoreObject, containers []v1.Container) errors.LintRuleError {
+	// https://regex101.com/library/FKV8ot
+	r := regexp.MustCompile("^[A-Fa-f0-9]{64}$")
+
 	for _, c := range containers {
-		imageParts := strings.Split(c.Image, ":")
-		if len(imageParts) != 2 {
+		if shouldSkipModuleContainer(object.Unstructured.GetName(), c.Name) {
+			continue
+		}
+
+		t, err := name.NewTag(c.Image)
+		if err != nil {
 			return errors.NewLintRuleError(
 				"CONTAINER003",
 				object.Identity()+"; container = "+c.Name,
 				nil,
-				"Can't parse an image for container",
+				"Can't parse an image for container: %v", err,
 			)
 		}
-		if imageParts[1] == "latest" {
+		registry := fmt.Sprintf("%s/%s", t.RegistryStr(), t.RepositoryStr())
+		tag := t.TagStr()
+
+		if registry != defaultRegistry {
 			return errors.NewLintRuleError("CONTAINER003",
 				object.Identity()+"; container = "+c.Name,
 				nil,
-				"Image tag \"latest\" used",
+				"All images must be deployed from the same default registry - "+defaultRegistry,
 			)
 		}
+
+		switch os.Getenv("CI") {
```

Comment:
Do we still need this branch if there is no dependency on the images_tags.json file?

Reply:
No, removed